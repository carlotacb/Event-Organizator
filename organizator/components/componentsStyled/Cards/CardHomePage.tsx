import React from "react";
// @ts-ignore
import styled from "styled-components/native";
import { FontAwesome5 } from "@expo/vector-icons";
import { parseDate } from "../../../utils/util-functions";
import {
  CardContainer,
  CardImage,
  CardText,
  CardTextContainer,
  CardTitle,
  TextLine,
} from "./Styles";
import AbsoluteChip from "../Chips/AbsoluteChip";

interface CardProps {
  // eslint-disable-next-line react/no-unused-prop-types
  id: string;
  title: string;
  startDate: string;
  endDate: string;
  location: string;
  headerImage: string;
  students: boolean;
}

const PastTag = styled.View`
  position: absolute;
  top: 25px;
  right: 25px;
  border: 4px solid #772323;
  background-color: rgba(119, 35, 35, 0.8);
  border-radius: 20px;
  padding: 5px 15px;
`;

const PastText = styled.Text`
  color: white;
  font-weight: bold;
  font-size: 20px;
`;

export default function CardHomePage(props: CardProps) {
  const { title, startDate, endDate, location, headerImage, students } = props;

  const isPast = () => startDate < new Date().toISOString();

  return (
    <CardContainer isPast={isPast()}>
      <CardImage source={{ uri: headerImage }} />
      <CardTextContainer>
        <CardTitle>{title}</CardTitle>
        <TextLine>
          <FontAwesome5 name="hourglass-start" size={16} />
          <CardText>{parseDate(startDate)}</CardText>
        </TextLine>
        <TextLine>
          <FontAwesome5 name="hourglass-end" size={16} />
          <CardText>{parseDate(endDate)}</CardText>
        </TextLine>
        <TextLine>
          <FontAwesome5 name="map-marker-alt" size={16} />
          <CardText>{location}</CardText>
        </TextLine>
        {isPast() && (
          <PastTag>
            <PastText>Past</PastText>
          </PastTag>
        )}
      </CardTextContainer>
      {students && (
        <AbsoluteChip name="Students only" backgroundColor="#b3f3f5" />
      )}
    </CardContainer>
  );
}
