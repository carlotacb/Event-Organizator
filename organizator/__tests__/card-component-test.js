import * as React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import CardHomePage from "../components/componentsStyled/Cards/CardHomePage";

it(`renders correctly a button`, () => {
  const tree = renderer
    .create(
      <CardHomePage
        title="Test event"
        startDate="2023-12-2200:36:41.562401TZ"
        endDate="2023-12-2117:27:38.081421TZ"
        location="UPC campus nord"
        id="this-is-a-test-id"
        headerImage="https://media.istockphoto.com/id/1087531642/vector/male-face-silhouette-or-icon-man-avatar-profile-unknown-or-anonymous-person-vector.jpg?s=612x612&w=0&k=20&c=FEppaMMfyIYV2HJ6Ty8tLmPL1GX6Tz9u9Y8SCRrkD-o%3D"
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
