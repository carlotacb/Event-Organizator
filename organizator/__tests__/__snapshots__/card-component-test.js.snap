// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly a button 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 20,
        "display": "flex",
        "flexDirection": "column",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "shadowColor": "rgba(0, 0, 0, 0.25)",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 10,
        "width": 280,
      },
    ]
  }
>
  <View>
    <View
      style={
        [
          {
            "borderRadius": 20,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": "100%",
            "opacity": 0.5,
            "width": "100%",
          },
        ]
      }
    >
      <Image
        source={
          {
            "uri": "https://media.istockphoto.com/id/1087531642/vector/male-face-silhouette-or-icon-man-avatar-profile-unknown-or-anonymous-person-vector.jpg?s=612x612&w=0&k=20&c=FEppaMMfyIYV2HJ6Ty8tLmPL1GX6Tz9u9Y8SCRrkD-o%3D",
          }
        }
        style={
          [
            {
              "borderRadius": 20,
              "height": 150,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "paddingBottom": 30,
              "paddingLeft": 25,
              "paddingRight": 25,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 20,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Test event
        </Text>
        <View
          style={
            [
              {
                "display": "flex",
                "flexDirection": "row",
                "gap": 10,
                "marginTop": 10,
              },
            ]
          }
        >
          <Text />
          <Text
            style={
              [
                {
                  "fontSize": 16,
                },
              ]
            }
          >
            NaN-NaN-NaN at NaN:NaN h
          </Text>
        </View>
        <View
          style={
            [
              {
                "display": "flex",
                "flexDirection": "row",
                "gap": 10,
                "marginTop": 10,
              },
            ]
          }
        >
          <Text />
          <Text
            style={
              [
                {
                  "fontSize": 16,
                },
              ]
            }
          >
            NaN-NaN-NaN at NaN:NaN h
          </Text>
        </View>
        <View
          style={
            [
              {
                "display": "flex",
                "flexDirection": "row",
                "gap": 10,
                "marginTop": 10,
              },
            ]
          }
        >
          <Text />
          <Text
            style={
              [
                {
                  "fontSize": 16,
                },
              ]
            }
          >
            UPC campus nord
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "rgba(119, 35, 35, 0.8)",
            "borderColor": "#772323",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 4,
            "paddingBottom": 5,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 5,
            "position": "absolute",
            "right": 15,
            "top": 15,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Past event
      </Text>
    </View>
  </View>
</View>
`;
